### PR TITLE
Draw throttle position in tuning tab

### LINKED ISSUE #1396: [FR] Show current throttle in PID Tuning tab
Currently, the PID tuning tab shows the curves for Pitch, Roll, Yaw and Throttle.

The graph for Pitch, Roll and Yaw shows the current stick position with a dot in the graph:

Capture

I would propose to add this to the Throttle graph as well, so a user can have a feel for the adjustments he / she might do

### PR DESCRIPTION
**Pull request by atomgomba**
This is aiming to fix #1396. Unfortunately, I don't have the needed hardware (RX+TX) with me at the moment to test this and so I'd really appreciate if someone could try it.

### CODE REVIEW

**Comment by IvoFPV** [comment]
You can select the MSP RX input on the configuration page and then go to the receiver tab and select Control sticks. That way you can test these things without a recevier.
Seems like it only goes from 50% up, meaning when throttle is 0%, dot is at 50%.

**Comment by atomgomba** [comment]
@IvoFPV Cool, I didn't know that! So now I can test and fix it, thanks!

**Comment by atomgomba** [comment]
Well, I assumed channel values go from 0 to 2000, when in fact the value range is 1000-2000. Should have known... :face_with_head_bandage:

File: src/js/tabs/pid_tuning.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ TABS.pid_tuning = {
     SETPOINT_WEIGHT_RANGE_LEGACY: 2.54,
     activeSubtab: 'pid',
     analyticsChanges: {},
+    THR_CHAN_INDEX: 3,
```
Comment (IvoFPV) [inline src/js/tabs/pid_tuning.js]: This can't be defined, what if I use TAER1234 channel map?

Comment (atomgomba) [reply src/js/tabs/pid_tuning.js]: Good point! I just looked at how the rate curve is drawn and there the values are simply referenced by index, like [here](https://github.com/betaflight/betaflight-configurator/blob/master/src/js/tabs/pid_tuning.js#L1716). And then in `fc.js` the array is defined [like this](https://github.com/betaflight/betaflight-configurator/blob/master/src/js/fc.js#L158). So this gave me the impression that the values are normalized by the firmware before they are sent over MSP.

Comment (IvoFPV) [reply src/js/tabs/pid_tuning.js]: It seems to be correct. I interpreted it wrong because of how msp rx works. 😕 This works as expected, my bad.

Comment (atomgomba) [reply src/js/tabs/pid_tuning.js]: Hm, so it seems the rate curve is also bugged, isn't it? It should use `RC_MAP` to determine the indices, if I'm not mistaken.

**Comment by atomgomba** [comment]
Btw, I think we have found a problem with the MSP Receiver. [This code](https://github.com/betaflight/betaflight-configurator/blob/master/src/js/tabs/receiver_msp.js#L14) seems to be problematic, since it's like it does not respect `RC_MAP`. Throttle in the MSP Receiver does not work when I switch to TAER. But that's a case for another PR. :)

**Comment by atomgomba** [comment]
> @atomgomba Point still gets moved on X axis when you change throttle mid, that shouldn't happen.

That's right, I fixed this now.

File: src/js/tabs/pid_tuning.js [hunk 2]
```diff
@@ -1833,6 +1880,17 @@ TABS.pid_tuning.checkRC = function() {
     return rateCurveUpdateRequired;
 };
 
+TABS.pid_tuning.checkThrottle = function() {
+    // Function monitors for change in the primary axes rc received data and returns true if a change is detected.
+    if (!this.oldThrottle) { 
+        this.oldThrottle = RC.channels[TABS.pid_tuning.THR_CHAN_INDEX];
```
Comment (IvoFPV) [inline src/js/tabs/pid_tuning.js]: 👾

File: src/js/tabs/pid_tuning.js [hunk 3]
```diff
@@ -1302,8 +1303,32 @@ TABS.pid_tuning.initialize = function (callback) {
...
+                return iT * iT * p1 + 2 * iT * t * p2 + t * t * p3;
+            }
+            
+            function getQuadraticCurvePoint(startX, startY, cpX, cpY, endX, endY, position) {
+                return {
+                    x:  getQBezierValue(position, startX, cpX, endX),
+                    y:  getQBezierValue(position, startY, cpY, endY)
+                };
+            }
+
```
Comment (IvoFPV) [inline src/js/tabs/pid_tuning.js]: 👾

File: src/js/tabs/pid_tuning.js [hunk 4]
```diff
@@ -1302,8 +1303,32 @@ TABS.pid_tuning.initialize = function (callback) {
...
+            /* 
+            Quadratic curve formula taken from: 
+                https://stackoverflow.com/a/9195706/176210 
+            */
+
+            function getQBezierValue(t, p1, p2, p3) {
+                var iT = 1 - t;
+                return iT * iT * p1 + 2 * iT * t * p2 + t * t * p3;
+            }
+
```
Comment (IvoFPV) [inline src/js/tabs/pid_tuning.js]: 👾

**Comment by atomgomba** [comment]
Done, I have cleaned the changed code. There was also a missing semicolon. I feel like I need to learn how to set up VSCode properly... All the other editors I'm using trim trailing whitespace automatically on changed lines, I bet VSCode could do the same somehow.

File: src/js/tabs/pid_tuning.js [hunk 5]
```diff
@@ -1302,8 +1302,32 @@ TABS.pid_tuning.initialize = function (callback) {
...
+            function getQuadraticCurvePoint(startX, startY, cpX, cpY, endX, endY, position) {
+                return {
+                    x:  getQBezierValue(position, startX, cpX, endX),
+                    y:  getQBezierValue(position, startY, cpY, endY),
+                };
+            }
+
+            /* --- */
+
+            setTimeout(function () {
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js]: What is the reason we are doing this asynchronously?

Comment (atomgomba) [reply src/js/tabs/pid_tuning.js]: Hm. It seems that the only reason is my misuse of the famous copy-paste technology. I edited the code so now it works more like in current master where calling `setTimeout()` did make sense. In there it is used to let the default HTML input validations trigger before calling the event handler.

Comment (mikeller) [reply src/js/tabs/pid_tuning.js]: Getting the triggers to work the way they should is a plus!

**Pull request merged**